### ISSUE TITLE
not seeing plot when i execute model.plot() 

### ISSUE THREAD

**Issue by Mawul4j** [SheffieldML/deepGPy#5]
I would like to reproduce the results in figure 4 of the paper "Nested Variational Compression in Deep Gaussian Processes" from "Nested Deep GPs.ipynb" notebook but model.plot() does not show any figure. Please what am i doing wrong?

**Comment by lawrennd** [comment]
@mzwiessele is this associated with the changes of the plotting code for
GPy?

**Comment by mzwiessele** [comment]
Could you provide a code snippet reproducing the error and your python version, so we can narrow it down? Also which GPy are you using?

**Comment by Mawul4j** [comment]
Okay, i have GPy (0.8.8, /home/mawulolo/GPy_Devel/GPy) installed locally.
Basically, model.plot() produces no figure.
and 
for layer in model.layers:
    layer.plot()
produces the following:
ValueError                                Traceback (most recent call last)
<ipython-input-18-1cf739f8d9ee> in <module>()
      1 for layer in model.layers:
----> 2     layer.plot()

/home/mawulolo/Desktop/Masters_Project/Lab_Session/layers.pyc in plot(self)
    329 
    330     def plot(self):
--> 331         plotting.plot_input_layer(self)
    332 
    333 

/home/mawulolo/Desktop/Masters_Project/Lab_Session/plotting.pyc in plot_input_layer(layer)
     67         Xnew, xmin, xmax = x_frame1D(layer.Z, resolution=200)
     68 
---> 69         sausage_plot(layer, Xnew, ax1)
     70         errorbars(layer, ax1)
     71         plt.setp(ax1.get_xticklabels(), visible=False)

/home/mawulolo/Desktop/Masters_Project/Lab_Session/plotting.pyc in sausage_plot(layer, Xnew, ax)
      8 def sausage_plot(layer, Xnew, ax):
      9     mu, var = layer.predict(Xnew)
---> 10     gpplot(Xnew, mu, mu + 2*np.sqrt(var), mu - 2*np.sqrt(var), ax=ax)
     11 
     12 def errorbars(layer, ax):

/home/mawulolo/GPy_Devel/GPy/GPy/plotting/matplot_dep/base_plots.pyc in gpplot(x, mu, lower, upper, edgecol, fillcol, ax, fignum, **kwargs)
     33 
     34     #here's the mean
---> 35     plots.append(meanplot(x, mu, edgecol, axes))
     36 
     37     #here's the box

/home/mawulolo/GPy_Devel/GPy/GPy/plotting/matplot_dep/base_plots.pyc in meanplot(x, mu, color, ax, fignum, linewidth, **kw)
     20 def meanplot(x, mu, color='#3300FF', ax=None, fignum=None, linewidth=2,**kw):
     21     _, axes = ax_default(fignum, ax)
---> 22     return axes.plot(x,mu,color=color,linewidth=linewidth,**kw)
     23 
     24 def gpplot(x, mu, lower, upper, edgecol='#3300FF', fillcol='#33CCFF', ax=None, fignum=None, **kwargs):

/home/mawulolo/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.pyc in inner(ax, *args, **kwargs)
   1809                     warnings.warn(msg % (label_namer, func.__name__),
   1810                                   RuntimeWarning, stacklevel=2)
-> 1811             return func(ax, *args, **kwargs)
   1812         pre_doc = inner.__doc__
   1813         if pre_doc is None:

/home/mawulolo/anaconda2/lib/python2.7/site-packages/matplotlib/axes/_axes.pyc in plot(self, *args, **kwargs)
   1425             kwargs['color'] = c
   1426 
-> 1427         for line in self._get_lines(*args, **kwargs):
   1428             self.add_line(line)
   1429             lines.append(line)

/home/mawulolo/anaconda2/lib/python2.7/site-packages/matplotlib/axes/_base.pyc in _grab_next_args(self, *args, **kwargs)
    384                 return
    385             if len(remaining) <= 3:
--> 386                 for seg in self._plot_args(remaining, kwargs):
    387                     yield seg
    388                 return

/home/mawulolo/anaconda2/lib/python2.7/site-packages/matplotlib/axes/_base.pyc in _plot_args(self, tup, kwargs)
    362             x, y = index_of(tup[-1])
    363 
--> 364         x, y = self._xy_from_xy(x, y)
    365 
    366         if self.command == 'plot':

/home/mawulolo/anaconda2/lib/python2.7/site-packages/matplotlib/axes/_base.pyc in _xy_from_xy(self, x, y)
    221         y = _check_1d(y)
    222         if x.shape[0] != y.shape[0]:
--> 223             raise ValueError("x and y must have same first dimension")
    224         if x.ndim > 2 or y.ndim > 2:
    225             raise ValueError("x and y can be no greater than 2-D")

ValueError: x and y must have same first dimension

**Comment by Mawul4j** [comment]
Please, is there anything i am doing wrong?

**Comment by Mawul4j** [comment]
I have managed to upgrade GPy to version 0.9.4 but this issue persists. 
Follow up: Is there any solution to this?

**Comment by mzwiessele** [comment]
I don't think it has anything to do with you doing something wrong. I cannot be sure though. Could you give us a minimalist example of when it goes wrong? So we can run it on newer versions of GPy and see how it performs there? Also we might be able to find the problem, if there is one in DeepGP.

Thanks

**Comment by Mawul4j** [comment]
Okay.
when i execute "model.plot()", basically it shows nothing 
Also when i execute 
for layer in model.layers:
----> layer.plot() 
ValueError: x and y must have same first dimension

I don't know if that makes sense.

**Comment by mzwiessele** [comment]
Do you have a piece of code, which we can run, which reproduces the error? That would be very helpful in order to see where the problem lies. 

Thanks

**Comment by Mawul4j** [comment]
[Nested.zip](https://github.com/SheffieldML/deepGPy/files/155044/Nested.zip)

**Comment by thangbui** [comment]
Have you tried commenting out this? https://github.com/SheffieldML/deepGPy/blob/master/step_fn_demo.py#L4

**Comment by Mawul4j** [comment]
Yes but it did not solve the problem for me. Did it work for you?

**Comment by thangbui** [comment]
OK nevermind my previous comment as you were trying to run the notebook, not the step function demo script.

I couldn't replicate the problem you have on a mac with the latest GPy.

**Comment by Mawul4j** [comment]
The plotting.py file contains code that only covers cases where input and output dimensions are ==1.
Could it be that the complete plotting.py file has not been uploaded here on github?

**Comment by jameshensman** [comment]
> Could it be that the complete plotting.py file has not been uploaded here
> on github?

No, if it works for Thang, it's on github.

If your output dimension is larger than 1, I suspect that simply no plot
will appear.

James.

**Comment by Mawul4j** [comment]
@jameshensman  Basically, i want to reproduce figure 4 of the paper "Nested Variational Compression in Deep Gaussian Processes". So, I ran the example in the notebook on robot wireless data to get that result.
Maybe i am doing this wrong, but can you please show me how figure 4 of that paper was produced with DeepGPy?
I believe Thang was helping with the step function demo.
